File: 3335-total-characters-in-string-after-transformations-i/3335-total-characters-in-string-after-transformations-i.cpp
class Solution {
public:
    int lengthAfterTransformations(string s, int t) {
        int mod = 1'000'000'007;
        vector<int> cnt(26, 0);

        for (const char& c: s)
            ++cnt[c - 97];

        for (int i = 0; i < t; ++i){
            vector<int> nxt(26, 0);
            nxt[0] = cnt[25];
            nxt[1] = (cnt[25] + cnt[0]) % mod;

            for (int j = 2; j < 26; ++j)
                nxt[j] = cnt[j - 1];
            cnt = nxt;
        }

        int ans = accumulate(cnt.begin(), cnt.end(), 0LL) % mod;
        return ans;
    }
};

File: 3335-total-characters-in-string-after-transformations-i/3335-total-characters-in-string-after-transformations-i.py
class Solution:
    def lengthAfterTransformations(self, s: str, t: int) -> int:
        n, mod = len(s), 10 ** 9 + 7
        cnt = [0] * 26
        
        for ch in s:
            cnt[ord(ch) - 97] += 1
        
        for _ in range(t):
            nxt = [0] * 26
            nxt[0] = cnt[25]
            nxt[1] = (cnt[25] + cnt[0]) % mod
            for i in range(2, 26):
                nxt[i] = cnt[i - 1]
            cnt = nxt
        ans = sum(cnt) % mod
        return ans
